security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticationResources.scala

```diff
@@ -5,15 +5,21 @@ import akka.http.scaladsl.server._
 import akka.http.scaladsl.server.directives.AuthenticationDirective
 import com.typesafe.config.Config
 import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport
-import pl.touk.nussknacker.security.AuthCredentials
+import pl.touk.nussknacker.security.AuthCredentials.PassedAuthCredentials
+import pl.touk.nussknacker.security.{AesCrypter, AuthCredentials, Crypter, NoOpCrypter}
 import sttp.client3.SttpBackend
-import sttp.tapir.EndpointInput
+import sttp.tapir.{EndpointInput, Mapping}
 
 import scala.concurrent.{ExecutionContext, Future}
 
 trait AuthenticationResources extends Directives with FailFastCirceSupport {
-  val name: String
-  val frontendStrategySettings: FrontendStrategySettings
+  def name: String
+
+  protected def frontendStrategySettings: FrontendStrategySettings
+
+  protected def rawAuthCredentialsMethod: EndpointInput[String]
+
+  protected def authCredentialsCrypter: Crypter = NoOpCrypter
```

Comment:
Let's remove this default, someone can introduce a dirty bug in their security extension and blame as for that. This class mixing two responsibilities: Handling common logic that is the same for all authentications and this specific for each authentication strategy. I think that we should start to split this. Eventually we should use some composition and keep in the interface only changing parts but we could start with moving this common things to the companion object. WDYT? Or maybe split it now? It was < 100 LOC before this change

Reply:
not relevant anymore